inductiva/generative/procedural/terrain.py

```diff
@@ -70,22 +70,48 @@ def generate_random_terrain(
                                          abs(percentile_translate_terrain))
     z_elevation += np.sign(percentile_translate_terrain) * percentile_translate
 
-    # Interpolate from the square grid to the desired grid.
-    x_square = np.linspace(*x_range, size_square)
-    y_square = np.linspace(*y_range, size_square)
+    z_elevation = interpolate_between_grids(x_range, y_range,
+                                            [size_square, size_square],
+                                            [x_num, y_num], z_elevation)
 
-    x_square, y_square = np.meshgrid(x_square, y_square, indexing="ij")
+    return z_elevation
 
-    x_grid = np.linspace(*x_range, x_num)
-    y_grid = np.linspace(*y_range, y_num)
+
+def create_grid(x_range, y_range, resolution):
+    """Create a grid of x and y values.
+
+    Given a [x_range, y_range] and a resolution for
+    each range, we return a meshgrid of x and y values.
+    That is, x_grid[i, j] = x_range[i] and
+    y_grid[i, j] = y_range[j].
+
+    Args:
+        x_range: The range of x values, in meters.
+        y_range: The range of y values, in meters.
+        resolution: Number of grid points in both
+            directions [x_res, y_res]
+    """
+
+    x_grid = np.linspace(*x_range, resolution[0])
```

Comment:
is this really "resolution"? A higher resolution means more detail. Does this parameter really mean resolution?
Does this correspond to the num parameter in linspace.
Could you please directly do the assignment num=resolution[0]

Are you starting to see the consistency problems that this code has?

Reply:
It can be considered resolution because with more res we also have more detail. But I will change to num_samples.

Reply:
use "num".
That way it becomes easier to understand since numpy is a sort of standard.